pkg/cmd/roachtest/test_runner.go

```diff
@@ -1140,7 +1150,10 @@ func (r *testRunner) runTest(
 		// Only include tests with a Run function in the summary output.
 		if s.Run != nil {
 			if t.Failed() {
-				r.status.fail[t] = struct{}{}
+				errWithOwner := failuresAsErrorWithOwnership(t.failures())
+				if errWithOwner != nil && !errWithOwner.InfraFlake {
```

Comment:
I don't think this is quite right. Whether a test failure is an infra flake can be determined by the predicate `errWithOwner != nil && errWithOwner.InfraFlake`.

In this case, we want to add to the `fail` set when `not infra-flake`. So what we want is (applying De Morgan's law): `errWithOwner == nil || !errWithOwner.InfraFlake`.

Reply:
Fixed